clis/mysql/tool.gpt

```diff
@@ -0,0 +1,24 @@
+name: mysql
+type: context
+description: Provides the ability to interact with mysql databases.
+share tools: sys.exec
+share credential: github.com/gptscript-ai/credential as cli.mysql.password with MYSQL_PWD as env and password as field and "Please enter your mysql user's password:" as message
```

Comment:
does this do what i think it does? im hoping this exposes the tool in such a way that it requires it for the consumer of this tool, but gives it a unique name on their system so that they could use this tool multiple times.

also, do we prefer writing a custom credential tool if we need multiple fields, or just having multiple credentials saved?

Reply:
As a side note, it's a little annoying to test these when we have such different behavior when things are local vs from git - locally, 'share credential' seems to do nothing.

Reply:
Yeah, `Share Credential` works the same way as `Share Tools`, `Share Context`, etc.
The tool that has `tools: github.com/gptscript-ai/tools/clis/mysql` will also have this credential tool run. The "unique name on their system" is just the credential alias that you specified here (`cli.mysql.password`).

If you need multiple fields, I would recommend just using the generic credential tool multiple times. It's more convenient.

I'm surprised if `share credential` truly does nothing locally. Maybe there were some mismatching file paths that were an issue? Idk. Were you initially trying to use a custom credential tool you wrote locally rather than the generic one?

Reply:
With this test script:
```
tools: github.com/drpebcak/tools/clis/mysql@mysql
chat: true
```
I would expect to get prompted to set up the credential when i call it, but it does not ask me to do anything. If i walk it through connecting to a db it just makes me type credentials into the chat session but those arent saved anywhere.

Reply:
for clarity - it only asks for credentials in the context of needing them to connect to the db, not because its running the credential tool.